extensions/http-curl/tests/HTTPUtils.h

```diff
@@ -0,0 +1,107 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef NIFI_MINIFI_CPP_HTTPUTILS_H
+#define NIFI_MINIFI_CPP_HTTPUTILS_H
+
+#if defined(_WIN32) || __cplusplus > 201103L
+#include <regex>
+#else
+#include <regex.h>
+#endif
+
+/**
+This function, unfortunately, assumes that we're parsing http components of a local host. On windows this is problematic
+so we convert localhost to our local hostname.
+  */
+inline bool parse_http_components(const std::string &url, std::string &port, std::string &scheme, std::string &path) {
+
+#if (__cplusplus > 201103L) || defined(_WIN32)
```

Comment:
I know it's old code, but we have regexutils, you can use that here. As we touch it, let's remove the ifdef magics.
It wraps regex and regex.h to have a common (RAII) interface, exactly for these scenarios.